pkg/controller/registry/resolver/fail_forward.go

```diff
@@ -0,0 +1,103 @@
+package resolver
+
+import (
+	"fmt"
+
+	operatorsv1 "github.com/operator-framework/api/pkg/operators/v1"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	v1listers "github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/listers/operators/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// IsFailForwardEnabled takes a namespaced operatorGroup lister and returns
+// True if an operatorGroup exists in the namespace and its upgradeStrategy
+// is set to UnsafeFailForward and false otherwise. An error is returned if
+// an more than one operatorGroup exists in the namespace.
+// No error is returned if no OperatorGroups are found to keep the resolver
+// backwards compatible.
```

Comment:
nit:
```suggestion
// IsFailForwardEnabled takes a namespaced operatorGroup lister and returns
// True if an operatorGroup exists in the namespace and its upgradeStrategy
// is set to UnsafeFailForward and false otherwise. An error is returned if
// more than one operatorGroup exists in the namespace. No error is returned 
// if no OperatorGroups are found to keep the resolver backwards compatible.
```